openedx/features/survey_report/queries.py

```diff
@@ -0,0 +1,72 @@
+"""
+Queries to get data from database.
+"""
+
+from datetime import datetime, timedelta
+
+from django.contrib.auth.models import User  # lint-amnesty, pylint: disable=imported-auth-user
+from django.db.models import Count, OuterRef, Q, Subquery
+
+from common.djangoapps.util.query import read_replica_or_default
+from common.djangoapps.student.models import CourseEnrollment
+from lms.djangoapps.grades.models import PersistentCourseGrade
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+def get_unique_courses_offered() -> int:
+    """
+    Get total number of unique course that started before today and have an open date,
+    or have not finished yet, whose number of enrollments is greater than 5.
+    """
+    return CourseOverview.objects.annotate(
+        count=Subquery(
+            CourseEnrollment.objects
+                            .filter(course_id=OuterRef('id'))
+                            .values('course_id')
+                            .annotate(count=Count('course_id'))
+                            .values('count')
+        ))\
+        .filter(count__gt=5)\
+        .filter(start__lt=datetime.now())\
+        .filter(Q(end__isnull=True) | Q(end__gt=datetime.now()))\
+        .using(read_replica_or_default())\
+        .count()
+
+
+def get_recently_active_users(weeks: int) -> int:
+    """
+    Get total number of users with last login in the last weeks.
+
+    Args:
+        weeks (int): number of weeks since the last login to considerate as an active learner.
+    """
+    return User.objects.filter(last_login__gte=datetime.now() - timedelta(weeks=weeks))\
```

Comment:
For each of these long queries, please output a log message before they start. That will make it easier for operators to know what's going on if it looks like it's just hanging for a long time while the query runs.